src/universal/modules/admin/containers/Impersonate/Impersonate.js

```diff
@@ -0,0 +1,110 @@
+import React, {PropTypes, Component} from 'react';
+import {connect} from 'react-redux';
+import {cashay} from 'cashay';
+import Helmet from 'react-helmet';
+import {withRouter} from 'react-router';
+import LoadingView from 'universal/components/LoadingView/LoadingView';
+import {showError} from 'universal/modules/notifications/ducks/notifications';
+import {setAuthToken} from 'universal/redux/authDuck';
+import {reset as resetAppState} from 'universal/redux/rootDuck';
+
+const impersonateTokenQuery = `
+query {
+  user @cached(type: "User") {
+    email
+    id,
+    jwt,
+    name,
+    picture
+  }
+}`;
+
+function createImposter(userId, dispatch, router) {
+  const variables = {userId};
+  cashay.mutate('createImposterToken', {variables}).then(async ({data, error}) => {
```

Comment:
@mattkrick is there a good way to avoid `then`-ing this so I may capture the error?